llama_stack/apis/shields/shields.py

```diff
@@ -19,34 +20,22 @@ class ShieldType(Enum):
     prompt_guard = "prompt_guard"
 
 
-class ShieldDef(BaseModel):
-    identifier: str = Field(
-        description="A unique identifier for the shield type",
-    )
-    shield_type: str = Field(
-        description="The type of shield this is; the value is one of the ShieldType enum"
-    )
-    params: Dict[str, Any] = Field(
-        default_factory=dict,
-        description="Any additional parameters needed for this shield",
-    )
-
-
 @json_schema_type
-class ShieldDefWithProvider(ShieldDef):
-    type: Literal["shield"] = "shield"
-    provider_id: str = Field(
-        description="The provider ID for this shield type",
-    )
+class Shield(Resource):
+    """A safety shield resource that can be used to check content"""
+
+    type: Literal[ResourceType.shield.value] = ResourceType.shield.value
+    shield_type: ShieldType
+    params: Dict[str, Any] = {}
 
 
 @runtime_checkable
 class Shields(Protocol):
     @webmethod(route="/shields/list", method="GET")
-    async def list_shields(self) -> List[ShieldDefWithProvider]: ...
+    async def list_shields(self) -> List[Shield]: ...
 
     @webmethod(route="/shields/get", method="GET")
-    async def get_shield(self, identifier: str) -> Optional[ShieldDefWithProvider]: ...
+    async def get_shield(self, identifier: str) -> Optional[Shield]: ...
 
     @webmethod(route="/shields/register", method="POST")
-    async def register_shield(self, shield: ShieldDefWithProvider) -> None: ...
+    async def register_shield(self, shield: Shield) -> None: ...
```

Comment:
I think it is more ergonomic to list the arguments individually because you may want to make some things optional (e.g., "name" and "provider_id") so I'd define it as:

```
register_shield(
  self, 
  identifier: str, # necessary 
  provider_id: Optional[str] = None, # if one, automatically picked 
  key: Optional[str] = None # defaults to identifier if not specified
)
```

Reply:
I think it will be cumbersome to specify like this especially since we also need shield type.

Reply:
updated.